aviary/mission/flops_based/phases/two_dof_phase.py

```diff
@@ -0,0 +1,267 @@
+
+        if meta_data is None:
+            meta_data = self.default_meta_data
+
+        self.meta_data = meta_data
+
+    def build_phase(self, aviary_options: AviaryValues = None):
+        '''
+        Return a new energy phase for analysis using these constraints.
+
+        If ode_class is None, default_ode_class is used.
+
+        If transcription is None, the return value from calling
+        make_default_transcription is used.
+
+        Parameters
+        ----------
+        aviary_options : AviaryValues (<empty>)
+            collection of Aircraft/Mission specific options
+
+        Returns
+        -------
+        dymos.Phase
+        '''
+        phase: dm.Phase = super().build_phase(aviary_options)
+
+        user_options: AviaryValues = self.user_options
+
+        throttle_setting = user_options.get_val('throttle_setting')
+        control_order = user_options.get_val('control_order')
+        opt = user_options.get_val('opt')
+
+        fix_initial = user_options.get_val('fix_initial')
+        initial_bounds = user_options.get_val('initial_bounds', units='ft')
+        initial_ref = user_options.get_val('initial_ref', units='ft')
+        duration_bounds = user_options.get_val('duration_bounds', units='ft')
+        duration_ref = user_options.get_val('duration_ref', units='ft')
+        ground_roll = user_options.get_val('ground_roll')
+        balance_throttle = user_options.get_val('balance_throttle')
+        rotation = user_options.get_val('rotation')
+        constraints = user_options.get_val('constraints')
+        optimize_mach = user_options.get_val('optimize_mach')
+        optimize_altitude = user_options.get_val('optimize_altitude')
+        mach_bounds = user_options.get_item('mach_bounds')
+        altitude_bounds = user_options.get_item('altitude_bounds')
+
+        if not balance_throttle:
+            phase.add_parameter(
+                Dynamic.Mission.THROTTLE,
+                opt=False,
+                units="unitless",
+                val=throttle_setting,
+                static_target=False)
+
+        if fix_initial:
+            phase.set_time_options(fix_initial=fix_initial, fix_duration=False,
+                                   units='ft', name=Dynamic.Mission.DISTANCE,
+                                   duration_bounds=duration_bounds, duration_ref=duration_ref)
+        else:
+            phase.set_time_options(fix_initial=fix_initial, fix_duration=False,
+                                   units='ft', name=Dynamic.Mission.DISTANCE,
+                                   initial_bounds=initial_bounds, initial_ref=initial_ref,
+                                   duration_bounds=duration_bounds, duration_ref=duration_ref)
+
+        phase.set_state_options("time", rate_source="dt_dr",
+                                fix_initial=fix_initial, fix_final=False, ref=100., defect_ref=100. * 1.e2, solve_segments='forward')
+
+        phase.set_state_options("mass", rate_source="dmass_dr",
```

Comment:
Upper bound on mass helped me with ipopt. I used the largest plane in the world as the upper.